cloudinit/distros/__init__.py

```diff
@@ -253,7 +253,7 @@ def _read_hostname(self, filename, default=None):
         raise NotImplementedError()
 
     @abc.abstractmethod
-    def _write_hostname(self, hostname, filename):
+    def _write_hostname(self, hostname, out_fn):
```

Comment:
I'd prefer we go the other direction with this one. In all of my previous programming contexts, "fn" has meant "function" and usually refers to some kind of function pointer. I think "fn" meaning "filename" in our codebase is confusing, and in general I think full names are better and more readable than abbreviations.

Reply:
Thanks for the review, I chose `out_fn` as it was used more commonly in the `_write_hostname()` definitions, but I see your point and I'm fine with `filename`. I'm updating the branch,